tree/tree/src/TChain.cxx

```diff
@@ -371,38 +372,10 @@ Int_t TChain::Add(const char* name, Long64_t nentries /* = TTree::kMaxEntries */
 
    // wildcarding used in name
    Int_t nf = 0;
-
-   Int_t slashpos = basename.Last('/');
-   TString directory;
-   if (slashpos>=0) {
-      directory = basename(0,slashpos); // Copy the directory name
-      basename.Remove(0,slashpos+1);      // and remove it from basename
-   } else {
-      directory = gSystem->UnixPathName(gSystem->WorkingDirectory());
-   }
-
-   const char *file;
-   const char *epath = gSystem->ExpandPathName(directory.Data());
-   void *dir = gSystem->OpenDirectory(epath);
-   delete [] epath;
-   if (dir) {
-      //create a TList to store the file names (not yet sorted)
-      TList l;
-      TRegexp re(basename,kTRUE);
-      while ((file = gSystem->GetDirEntry(dir))) {
-         if (!strcmp(file,".") || !strcmp(file,"..")) continue;
-         TString s = file;
-         if ( (basename!=file) && s.Index(re) == kNPOS) continue;
-         l.Add(new TObjString(file));
-      }
-      gSystem->FreeDirectory(dir);
-      //sort the files in alphanumeric order
-      l.Sort();
-      TIter next(&l);
-      TObjString *obj;
+   try {
```

Comment:
the `try` seems to encompass too much code, can it include just the `ExpandGlob` call?